tests/linen/linen_module_test.py

```diff
@@ -1382,6 +1381,170 @@ class MyComponent2(Generic[T], nn.Module):
     class MyModule2(nn.Module):
       submodule: MyComponent2[jnp.ndarray]
 
+  def test_module_summary(self):
+    """
+    This test creates a Table using `module_summary` and checks that it
+    matches the expected output given the CNN model defined in `_get_tabulate_cnn`.
+    """
+    CNN = _get_tabulate_cnn()
+    batch_size = 32
+
+    x = jnp.ones((batch_size, 28, 28, 1))
+    module = CNN()
+
+    table = module_summary(
+      module, 
+      {"dropout":random.PRNGKey(0), "params": random.PRNGKey(1)}, 
+      x, 
+      training=True,
+    )
+
+    # 11 rows = 1 Inputs + 4 ConvBlock_0 + 4 ConvBlock_1 + 1 Dense_0 + 1 Module output
+    assert len(table) == 11
```

Comment:
Pytest somehow overrides `assert` such that if you change this `11` to a `12` the error message when you run this test like this:

```bash
pytest tests/linen/summary_test.py
```
is presented in a rich form as:
```
>     assert len(table) == 12
E     AssertionError: assert 11 == 12
E      +  where 11 = len([Row(path=('Inputs',), outputs=(DeviceArray([[[[1.],\n               [1.],\n               [1.],\n               ...,\n   ...2), {'a': ShapeDtypeStruct(shape=(32, 10), dtype=float32), 'b': ShapeDtypeStruct(shape=(32, 10), dtype=float32)}),)}})])
```
Using `assertEquals` looks like this:
```
>     self.assertEqual(len(table), 12)
E     AssertionError: 11 != 12
```

Reply:
Ok, I see `assertEqual` is used throughout the code base so I will change it to match the style.

Reply:
I do prefer the `assert` style as it looks more natural.

Reply:
`self.assertEqual` provides a bit more functionality than just `assert`. It provides better error messages in the case of failure, and it will automatically call `assertListEqual`, `assertDictEqual`, ... if the two inputs have the same type.